lib/services/profiles-api-service.js

```diff
@@ -81,6 +84,14 @@ function profileApiServiceFactory(
                         query.mac
                     );
                 }
+            }).then(function () {
+                if ( req.get(Constants.HttpHeaders.ApiProxyIp) ) {
+                    var proxy =  'http://%s:%s'.format(
+                        req.get(Constants.HttpHeaders.ApiProxyIp),
+                        req.get(Constants.HttpHeaders.ApiProxyPort)
+                    );
+                    return waterline.lookups.upsertProxyToMacAddress(proxy, query.mac);
+                }
```

Comment:
@shanemsully Yeah - I think that's the behavior we want.  It turns out that the profiles API doesn't look at the fulfillment value of the returned promise, so it's not an important detail at the moment but I think we should keep things consistent with the previous implementation.